src/ui/public/kuery/suggestions/value.js

```diff
@@ -0,0 +1,48 @@
+import { flatten, memoize } from 'lodash';
+import chrome from 'ui/chrome';
+import { escapeQuotes } from './escape_kuery';
+
+const baseUrl = chrome.addBasePath('/api/kibana/suggestions/values');
+const type = 'value';
+
+export function getSuggestionsProvider({ $http, config, indexPatterns }) {
+  const allFields = flatten(indexPatterns.map(indexPattern => indexPattern.fields.raw));
+  const shouldSuggestValues = config.get('filterEditor:suggestValues');
+  return shouldSuggestValues ? memoize(getValueSuggestions, getFieldQueryHash) : () => Promise.resolve([]);
```

Comment:
I like this, but I think adding some kind of TTL here would be a good idea. The cache only gets cleared when users refresh the browser or navigate to an app outside of core, which could be never, so new indexed values won't show up. 

A TTL of, say, a minute, would calm the lookup requests to ES while also allowing new values to show up over time.